server/rest/src/main/java/org/infinispan/rest/resources/ContainerResource.java

```diff
@@ -599,4 +644,30 @@ public CompletionStage<Void> onConfigurationEvent(ConfigurationChangedEvent even
          return eventStream.sendEvent(sse);
       }
    }
+
+   @Immutable
+   private class LifecycleEvent implements JsonSerialization, Serializable {
```

Comment:
Class can be `static`.

Reply:
Is there any reason why this implements `Serializable`? In general we avoid this as we rely on ProtoStream/Externalizers for internal marshalling requirements.

Reply:
Guess it was just habit in adding the `Serializable`. Updating.